docs/GettingStarted.md

```diff
@@ -0,0 +1,139 @@
+
+```c++
+constexpr int kDmaAddress = 0xa0010000;
+auto dma = platform->GetDataMover(kDmaAddress);
+```
+
+where `kDmaAddress` is the address of the DMA instance we want to control. This is given by the design.
+
+4) Create the IP core instances to interact with them.
+
+```c++
+constexpr int kAccelAddress = 0xa0000000;
+auto accel = platform->GetAccelerator(kAccelAddress);
+```
+
+where `kAccelAddress` is the AXI4-Lite control port of the IP core. As a requirement:
+
+> The IP Cores must have an AXI4-Lite port as a control port
+
+5) Get buffers to exchange data. These buffers are usually dual memory: they are mapped into host and device regions (physically contiguous).
+
+```c++
+std::size_t input_size = 64; // bytes
+std::size_t output_size = 64; // bytes
+auto type = MemoryType::Dual; // dual memory
+
+auto in_mem = dma->GetBuffer(input_size, type);
+auto out_mem = dma->GetBuffer(output_size, type);
+```
+
+where the `GetBuffer()` method includes: `size` in bytes and `type` of the memory type:
+
+* `cynq::MemoryType::Dual`: allocates two memory regions, one accessible from host and another from device.
+* `cynq::MemoryType::Cacheable`: allocates a memory region which is cacheable.
+* `cynq::MemoryType::Host`: allocates a host-only memory.
+* `cynq::MemoryType::Device`: allocates a device-only memory.
+
+6) To access the data from the memory buffers, you can use the `HostAddress` method.
+
+```c++
+using DataType = uint64_t;
+DataType* A = in_mem->HostAddress<DataType>().get();
+
+A[5] = 1;
+```
+
+The `HostAddress<T>()` maps the memory into a pointer that is accessible to the host. `T` can be any type that can be reinterpretedly casted.
+
+7) Write/Read the IP Core / Accelerator registers. You can use the `Write()` and `Read()` methods.
+
+```c++
+uint16_t input_a_cols;
+
+// Read a single element
+accel->Read(0x20, &input_a_cols, 1);
+
+// Write a single element
+input_a_cols = 64;
+accel->Write(0x28, &input_a_cols, 1);
+```
+
+Both `Read(addr, data*, elems)` and `Write(addr, data*, elems)` have the same arguments:
+
+* `addr`: offset address of the register
+* `data*`: data pointer to be read/written
+* `elems`: number of elements of type `data` to read/write
+
+8) Start/Stop the accelerator by writing the control register
+
+```c++
+accel->Start(StartMode::Continuous);
+
+accel->Stop();
+```
+
+To start the accelerator, you can use the `Start()` method, which receives either of the following values:
+
+* `cynq::StartMode::Once`: turns on the accelerator 
+* `cynq::StartMode::Continuous`: turns on the accelerator in autorestart mode.
+
+9) Transferring information requires the synchronisation of the memory buffers and the interaction with the data mover / DMA.
+
+To synchronise the buffers, it is possible to use `Sync()` method: 
+
+```c++
+in_mem->Sync(SyncType::HostToDevice);
+out_mem->Sync(SyncType::DeviceToHost);
+```
+
+that takes any of the following values:
```

Comment:
any can be interpreted as a list of values, clarify that only ONE value is passed to the method but it is ONE from the list provided.